runtime/src/main/java/org/corfudb/runtime/collections/StreamingManager.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    /**
+     * Subscribe to transaction updates.
+     *
+     * @param streamListener   client listener for callback
+     * @param namespace        namespace of interested tables
+     * @param streamTag        only updates of tables with the stream tag will be polled
+     * @param tablesOfInterest only updates from these tables will be returned
+     * @param lastAddress      last processed address, new notifications start from lastAddress + 1
+     */
+    synchronized void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                                @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                                long lastAddress) {
+        subscribe(streamListener, namespace, streamTag, tablesOfInterest, lastAddress, DEFAULT_BUFFER_SIZE);
+    }
+
+    /**
+     * Subscribe to transaction updates.
+     *
+     * @param streamListener   client listener for callback
+     * @param namespace        namespace of interested tables
+     * @param streamTag        only updates of tables with the stream tag will be polled
+     * @param tablesOfInterest only updates from these tables will be returned
+     * @param lastAddress      last processed address, new notifications start from lastAddress + 1
+     * @param bufferSize       maximum size of buffered transaction entries
+     */
+    synchronized void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                                @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                                long lastAddress, int bufferSize) {
+        if (bufferSize < 1) {
+            throw new IllegalArgumentException("subscribe: Buffer size cannot be less than 1.");
+        }
+
+        if (subscriptions.containsKey(streamListener)) {
+            // Multiple subscribers subscribing to same namespace and table is allowed
+            // as long as the hashcode() and equals() method of the listeners are different.
+            throw new StreamSubscriptionException(
+                    "StreamingManager::subscribe: listener already registered " + streamListener);
+        }
+
+        StreamingMetrics metrics = new StreamingMetrics(
+                streamListener, namespace, streamTag, runtime.getParameters().getMetricRegistry());
+        StreamSubscription subscription = new StreamSubscription(
+                runtime, streamListener, namespace, streamTag, tablesOfInterest, bufferSize, metrics);
+        subscriptions.put(streamListener, subscription);
+
+        pollingExecutor.submit(new StreamPollingTask(this, lastAddress, subscription, pollingExecutor));
+        notificationExecutor.submit(new StreamNotificationTask(this, subscription, notificationExecutor));
+
+        log.info("Subscribed stream listener {}, numSubscribers: {}, streamTag: {}, lastAddress: {}, " +
```

Comment:
Currently if user overloads toString(), then we have a meaningful identifier, and that's what we currently do in testing. I'm trying to avoid changing interfaces in this patch, but sure later we could change the interface to have better debuggability.